scripts/generateTranslations.ts

```diff
@@ -0,0 +1,454 @@
+                            return node;
+                        }
+
+                        // Recursively translate all complex template spans first
+                        const translatedComplexSpans = new Map<number, ts.TemplateSpan>();
+
+                        // Template expression is complex:
+                        for (const span of node.templateSpans) {
+                            if (this.isSimpleTemplateSpan(span)) {
+                                continue;
+                            }
+                            const hash = StringUtils.hash(span.expression.getText());
+                            const translatedSpan = ts.visitNode(span, visit);
+                            translatedComplexSpans.set(hash, translatedSpan as ts.TemplateSpan);
+                        }
+
+                        // Build the translated template expression, referencing the translated template spans as necessary
+                        return this.stringToTemplateExpression(translatedTemplate, translatedComplexSpans);
+                    }
+                }
+                return ts.visitEachChild(node, visit, context);
+            };
+
+            return (node: ts.SourceFile) => {
+                const transformedNode = ts.visitNode(node, visit) ?? node; // Ensure we always return a valid node
+                return transformedNode as ts.SourceFile; // Safe cast since we always pass in a SourceFile
+            };
+        };
+    }
+}
+
+/**
+ * The main function mostly contains CLI and file I/O logic, while TS parsing and translation logic is encapsulated in TranslationGenerator.
+ */
+async function main(): Promise<void> {
+    const config = {
+        flags: {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            'dry-run': {
+                description: 'If true, just do local mocked translations rather than making real requests to an AI translator.',
+            },
+        },
+        namedArgs: {
+            locales: {
+                description: 'Locales to generate translations for.',
+                default: DEFAULT_LOCALES_TO_TRANSLATE.join(','),
+                parse: (val: string) => {
+                    const locales = val.split(',');
+                    for (const locale of locales) {
+                        // Note: we are making a temporary exception for `it` for the sake of testing this script.
+                        // We can remove that check once we add `it` and other locales to CONST.LOCALES.
+                        if (!LocaleUtils.isValidLocale(locale) && locale !== 'it') {
+                            throw new Error(`Invalid locale ${locale}`);
+                        }
+                    }
+                    return locales as Locale[];
+                },
+            },
+        },
+    } as const;
+    const cli = new CLI<(typeof config)['flags'], (typeof config)['namedArgs'], undefined>(config);
```

Comment:
Couldn't figure out why type inference wasn't working as expected here without explicit generics. I asked @blazejkustra and @fabioh8010 for help figuring it out, but I don't think that needs to block this PR.